### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
List of changes:
- fix typos and markdown issues
- fix missing embedded readme in module doc
- resolve warnings on missing function in changelog
- generate html only
- move ex_doc options into own function in project config

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-
 # Goth
 
-<!-- MDOC !-->
+![CI](https://github.com/peburrows/goth/workflows/CI/badge.svg)
+
+<!-- MDOC -->
 
-Google + Auth = Goth
+__Go__ogle + Au__th__ = Goth
```
Comment (wojtekmach) [inline README.md:7]: this is clever but let's keep it simple. :)

```suggestion
Google + Auth = Goth
```

File: README.md [hunk 2]
```diff
@@ -29,7 +31,10 @@ A simple library to generate and retrieve OAuth2 tokens for use with Google Clou
      use Application
 
      def start(_type, _args) do
-       credentials = "GOOGLE_APPLICATION_CREDENTIALS_JSON" |> System.fetch_env!() |> Jason.decode!()
+       credentials =
+         "GOOGLE_APPLICATION_CREDENTIALS_JSON"
+         |> System.fetch_env!()
+         |> Jason.decode!()
```
Comment (wojtekmach) [inline README.md:37]:
```suggestion
         |> Jason.decode!()

```

File: README.md [hunk 3]
```diff
@@ -1,16 +1,18 @@
-![CI](https://github.com/peburrows/goth/workflows/CI/badge.svg)
-
 # Goth
 
-<!-- MDOC !-->
+![CI](https://github.com/peburrows/goth/workflows/CI/badge.svg)
+
+<!-- MDOC -->
```
Comment (wojtekmach) [inline README.md:5]: Given we include readme in hexdocs, let's skip it from moduledoc. I think that's what I had in mind but I forgot to remove the MDOCs.

```suggestion
```

File: README.md [hunk 4]
```diff
@@ -89,7 +95,7 @@ A simple library to generate and retrieve OAuth2 tokens for use with Google Clou
 
 See `Goth.start_link/1` for more information about possible configuration options.
 
-<!-- MDOC !-->
+<!-- MDOC -->
```
Comment (wojtekmach) [inline README.md:98]:
```suggestion
```

File: UPGRADE_GUIDE.md [hunk 5]
```diff
@@ -1,9 +1,10 @@
...
 
-Earlier versions of Goth relied on global application environment configuration which is deprecated
-in favour of a more direct and explicit approach in Goth v1.3+. Previously, Goth depended on on
-the HTTPoison HTTP client, but now it has an _optional_ dependency on Hackney. Thus, new projects
-using Goth and wanting to use Hackney will need to explicitly include it in the dependencies.
+Earlier versions of Goth relied on global application environment configuration
+which is deprecated in favour of a more direct and explicit approach in Goth
+v1.3+. Previously, Goth depended on on the HTTPoison HTTP client, but now it
+has an _optional_ dependency on Hackney. Thus, new projects using Goth and
+wanting to use Hackney will need to explicitly include it in the dependencies.
```
Comment (wojtekmach) [inline UPGRADE_GUIDE.md:7]:
```suggestion
v1.3+.
```

File: lib/goth.ex [hunk 6]
```diff
@@ -1,9 +1,8 @@
 defmodule Goth do
   @external_resource "README.md"
-
-  @moduledoc """
-  A Goth token server.
-  """
+  @moduledoc File.read!("README.md")
+             |> String.split("<!-- MDOC -->")
+             |> Enum.fetch!(1)
```
Comment (wojtekmach) [inline lib/goth.ex:5]:
```suggestion
  @moduledoc """
  A Goth token server.
  """
```

**Comment by wojtekmach** [comment]
Thank you!

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**